core/src/main/java/io/grpc/NameResolverProvider.java

```diff
@@ -115,26 +124,41 @@ public String getDefaultScheme() {
     }
 
     private void checkForProviders() {
-      Preconditions.checkState(!providers.isEmpty(),
-          "No NameResolverProviders found via ServiceLoader, including for DNS. "
-          + "This is probably due to a broken build. If using ProGuard, check your configuration");
+      if (providers.isEmpty()) {
+        String msg = "No NameResolverProviders found via ServiceLoader, including for DNS. "
+            + "This is probably due to a broken build. If using ProGuard, check your configuration";
+        throw new UnsupportedOperationException(msg);
```

Comment:
I would prefer to leave the exception as is, since UnsupportedOperationException implies to me that the class as defined does not support this operation at all.

Reply:
ISE has a specific meaning: methods were called out of order.   This is for calling read() after close()  and patterns like that.  I'm open to another Exception, maybe just plain RuntimeException?

Reply:
RuntimeException would be ok. This change probably won't cause problems for users but worth documenting in the release notes.

Reply:
Done.